experimental/dds/tree/src/SharedTree.ts

```diff
@@ -89,6 +97,52 @@ import { IdCompressor, createSessionId } from './id-compressor';
 import { convertEditIds } from './IdConversion';
 import { MutableStringInterner } from './StringInterner';
 
+/**
+ * The write format and associated options used to construct a `SharedTree`
+ * @public
+ */
+export type SharedTreeArgs<WF extends WriteFormat = WriteFormat> = [writeFormat: WF, options?: SharedTreeOptions<WF>];
+
+/**
+ * The type of shared tree options for a given write format
+ * @public
+ */
+export type SharedTreeOptions<WF extends WriteFormat> = WF extends WriteFormat.v0_0_2
+	? SharedTreeOptions_0_0_2
+	: WF extends WriteFormat.v0_1_1
+	? SharedTreeOptions_0_1_1
+	: never;
+
+/**
+ * Configuration options for a SharedTree with write format 0.0.2
+ * @public
+ */
+export interface SharedTreeOptions_0_0_2 {
+	/** Determines if the history is included in summaries */
+	summarizeHistory?: boolean;
```

Comment:
Make it difficult to set this flag, and comment that if you summarize history, it incurs a performance cost and can never be deleted